tests/test_daemon.cpp

```
@@ -129,6 +129,8 @@ struct Daemon : public mpt::DaemonTestFixture
                              a few more tests for `false`, since there are different portions of code depending on it */
         EXPECT_CALL(mock_settings, get(Eq(mp::winterm_key))).WillRepeatedly(Return("none"));
         EXPECT_CALL(mock_settings, get(Eq(mp::bridged_interface_key))).WillRepeatedly(Return("eth8"));
+        EXPECT_CALL(mock_settings, get(Eq(mp::driver_key)))
+            .Times(AnyNumber()); // TODO lxd and libvirt migration, remove
```

Should this be `.WillRepeatedly(Return("qemu"))`? Otherwise it would return empty strings.

---

It is basically the same effect for now. The point of the `EXPECT_CALL` here is just matching the `MP_SETTINGS.get(mp::driver_key);` call in `lxd_and_libvirt_deprecation_warning` and make it skip the branch. Both empty string and "qemu" can do it.

---

But this will match *any* attempt to obtain the driver (present and future) and silently return an invalid value. Better prevent that, don't you think?

---

Yep, that is right? `.WillRepeatedly(Return("qemu"))` is more future proof.